spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java

```diff
@@ -184,10 +193,17 @@ public DataSourceProperties cloudSqlDataSourceProperties(GcpCloudSqlProperties g
 				properties.setUrl(cloudSqlJdbcInfoProvider.getJdbcUrl());
 			}
 			else {
-				LOGGER.warn("spring.datasource.url is specified. Not using generated Cloud SQL configuration");
+				LOGGER.warn("spring.datasource.url is specified. "
+						+ "Not using generated Cloud SQL configuration");
 			}
 
-			setCredentialsProperty(gcpCloudSqlProperties, gcpProperties);
+			if (environment.getProperty(
+					GcpCloudFoundryEnvironmentPostProcessor.VCAP_SERVICES_ENVVAR) != null) {
```

Comment:
I think we have that `credentials.getEncodedKey()` to consult here.
Let's try do not pull any CF specific logic into the target configurations!